miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -478,6 +530,33 @@ private void setIdentity(DetailedSample descendant) throws IOException, MisoNami
     child.setParent(identity);
   }
 
+  /**
+   * Finds the identity of a sample by climbing the hierarchy
+   * 
+   * @param sample
+   * @return identity attached to sample
+   * @throws IOException
+   */
+  private SampleIdentity getIdentity(DetailedSample sample) throws IOException {
+    if (sample.getParent() == null) {
+      throw new IllegalArgumentException(ERR_MISSING_PARENT_ID);
+    }
+    DetailedSample child = sample;
+    DetailedSample parent = sample.getParent();
+
+    while (parent.getParent() != null) {
+      child = parent;
+      parent = child.getParent();
+    }
+
+    if (!isIdentitySample(parent)) {
+      throw new IllegalStateException("Missing Identity at root of hierarchy");
+    }
+
+    SampleIdentity identity = (SampleIdentity) parent;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "identity". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)